pkg/authentication/cookies.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2024 Canonical Ltd.
+// SPDX-License-Identifier: AGPL-3.0
+
+package authentication
+
+import (
+	"net/http"
+	"time"
+)
+
+var (
+	EPOCH             = time.Unix(0, 0)
+	NONCE_COOKIE_NAME = "nonce"
+)
+
+func SetNonceCookie(w http.ResponseWriter, nonce string) {
+	expires := time.Now().Add(2 * time.Minute)
+	http.SetCookie(w, &http.Cookie{
+		Name: NONCE_COOKIE_NAME,
+		// TODO @barco: encrypt this value when cookie encryption is added to the codebase
+		Value:    nonce,
+		Path:     "/api/v0/login",
```

Comment:
I may be wrong, but I am not sure if this will work when running under a path, e.g. the login URL may be:
`https://example.com/iam-admin/api/v0/login`, afaict in that case this cookie path won't work

Reply:
You're not wrong. Right now every handler endpoint is hardcoded and does not take into account any prefix. When we implement that, we will have to take care of that in all the codebase.

Reply:
ingress controller will take care of doing the path rewriting, now do we actually care in this case what s inside the cookie given that the path of the request will be rewritten to what we know?

Reply:
We could "ignore" the path although it's nice to keep this cookie "scoped" to the only endpoint that needs it.
My proposal is: leave it as is. With cookie encryption implemented, we would have our secret key as an additional strong layer of security and could decide to relax the `path` constraint if it troubles us with the deployment